spring-integration-ftp/src/main/java/org/springframework/integration/ftp/inbound/FtpInboundFileSynchronizer.java

```diff
@@ -42,6 +43,20 @@ public FtpInboundFileSynchronizer(SessionFactory<FTPFile> sessionFactory) {
 		super(sessionFactory);
 	}
 
+	@Override
+	public void setRemoteDirectoryExpression(Expression remoteDirectoryExpression) {
+		if (remoteDirectoryExpression != null) {
+			super.setRemoteDirectoryExpression(remoteDirectoryExpression);
+		}
+	}
+
+	@Override
+	public void afterPropertiesSet() {
+		if (getRemoteDirectoryExpression() == null) {
+			setRemoteDirectory(null);
+		}
+		super.afterPropertiesSet();
+	}
```

Comment:
OK; this then? ...

	public FtpInboundFileSynchronizer(SessionFactory<FTPFile> sessionFactory) {
		super(sessionFactory);
		initDefaults();
	}

	private void initDefaults() {
		super.setRemoteDirectoryExpression(new LiteralExpression(null));
	}

You can't call the super method directly from the ctor because it's not final.